File: MicroServices/MicroServices/referenciel-service/src/main/java/com/bfi/referentiel/service/ContactService.java
package com.bfi.referentiel.service;

import com.bfi.referentiel.common.dto.ContactDTO;

import java.util.List;

public interface ContactService {

    void addContact(ContactDTO contact);

    void updateContact(ContactDTO contact);

    void deleteContact(Long id);

    List<ContactDTO> getAllContacts();

    List<ContactDTO> findContactByTierId(Long id);
}

File: CartagoFrontV2-master/src/app/services/referenciel/contact.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Contact } from '../../models/contact';
@Injectable({
  providedIn: 'root',
})
export class ContactService {
  constructor(private http: HttpClient) {}

  getAllContacts(): Observable<Contact[]> {
    return this.http.get<Contact[]>('http://localhost:8080/Contacts');
  }

  getContactByTierId(id: number): Observable<Contact[]> {
    return this.http.get<Contact[]>(
      'http://localhost:8080/Contacts/' + id + '/Tier'
    );
  }

  addContact(contact: Contact): Observable<Contact> {
    return this.http.post<Contact>('http://localhost:8080/Contacts/', contact);
  }

  updateContact(contact: Contact): Observable<Contact> {
    return this.http.put<Contact>('http://localhost:8080/Contacts/', contact);
  }

  deleteContact(id: number): Observable<Contact> {
    return this.http.delete<Contact>('http://localhost:8080/Contacts/' + id);
  }
}
